src/markdown_generators.js

```diff
@@ -162,31 +163,88 @@ const generateMarkdownFromJson = courseData => {
   let courseHomeMarkdown = generateCourseHomeFrontMatter(courseData)
   courseHomeMarkdown += generateCourseFeatures(courseData)
   courseHomeMarkdown += generateCourseCollections(courseData)
-  const coursePagesWithText = courseData["course_pages"].filter(
-    page => page["text"]
+  const rootSections = courseData["course_pages"].filter(
+    page =>
+      page["parent_uid"] === courseData["uid"] &&
+      page["type"] !== "CourseHomeSection"
   )
+  let menuIndex = 0
   return [
     {
       name: "_index.md",
       data: courseHomeMarkdown
     },
-    ...coursePagesWithText.map((page, menuIndex) => {
+    ...rootSections.map(page => {
       const pageName = page["short_url"]
       let courseSectionMarkdown = generateCourseSectionFrontMatter(
         page["title"],
-        page["short_url"],
+        `${page["uid"]}_${page["short_url"]}`,
+        null,
         (menuIndex + 1) * 10,
         courseData["short_url"]
       )
+      menuIndex++
       courseSectionMarkdown += generateCourseSectionMarkdown(page, courseData)
+      const children = courseData["course_pages"].filter(
+        childPage => childPage["parent_uid"] === page["uid"]
+      )
       return {
-        name: `sections/${pageName}.md`,
-        data: courseSectionMarkdown
+        name:
+          children.length > 0
+            ? `sections/${pageName}/_index.md`
+            : `sections/${pageName}.md`,
+        data:     courseSectionMarkdown,
+        children: generateChildMarkdownRecursive(
+          page,
+          children,
+          menuIndex,
+          courseData
+        )
       }
     })
   ]
 }
 
+const generateChildMarkdownRecursive = (
```

Comment:
it would be sort of nice aesthetically if we could map one function across `rootSections` above and also map that across the children - I see some logic difference in this function in terms of how we calculate paths and stuff, but do you think it could be combined with the above into one recursive function to traverse all the pages? if not no worries, was just thinking it would be nice to be able to represent this transformation as a single function.

Reply:
I think I was able to refactor this nicely so it's just a function that can be passed into map on any array of `course_pages` objects.